cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -777,12 +777,34 @@ private String getRackHandleNull(Node node) {
     return allBrokers;
   }
 
+  /**
+   * Get all the dead brokers in the cluster based on the partition assignment. If a metadata refresh failed due to
+   * timeout, the current metadata information will be used. This is to handle the case that all the brokers are down.
+   * @param timeout the timeout in milliseconds.
+   * @return All the dead brokers in the cluster.
+   */
+  public Set<Integer> deadBrokers(long timeout) {
```

Comment:
I believe both this function and `deadBrokers(Cluster kafkaCluster)` can be named slightly more correctly. They actually return dead brokers with replicas; hence, if there is a dead broker with no replica, these functions will not return such a broker.

Can we rename these functions to `deadBrokersWithReplicas`?